src/ripple/overlay/Overlay.h

```diff
@@ -113,6 +113,18 @@ class Overlay
     PeerSequence
     getActivePeers () = 0;
 
+    /** Calls the checkSanity function on each peer
+    */
+    virtual
+    void
+    checkSanity (std::uint32_t index) = 0;
```

Comment:
@JoelKatz Overlay.h includes Peer.h, and the Peer interface implicitly refers to protocol-level primitives like `LedgerIndex` (although it doesn't use the type, it uses `std::uint32_t`). So its completely reasonable for Overlay.h to include ripple/protocol/Protocol.h to get access to basic types. From a design standpoint it does follow that both Overlay and Peer would offer interfaces that deal in protocol types. So its okay to add the include.